src/term_ctl.c

```diff
@@ -17,64 +17,203 @@
 
 #include "term_ctl.h"
 
+#ifdef _WIN32
+#include <io.h>
+#include <limits.h>
+#include <windows.h>
+
+#ifndef STDOUT_FILENO
+#define STDOUT_FILENO   1
+#endif
+
+#ifndef STDERR_FILENO
+#define STDERR_FILENO   2
+#endif
+
+static CONSOLE_SCREEN_BUFFER_INFO console_info;
+static BOOL                       console_redirected = FALSE;
+static HANDLE                     console_hnd = NULL;
+static FILE                      *console_file = NULL;
+static WORD                       active_fg, active_bg;
```

Comment:
Can you pull this from wAttributes when needed?

Reply:
I see. We would need to set fg and bg with a single call then, or use this static.